File: src/main/java/calebzhou/rdi/core/server/command/impl/InspectCommand.java
package calebzhou.rdi.core.server.command.impl;

import calebzhou.rdi.core.server.command.RdiCommand;
import calebzhou.rdi.core.server.utils.RdiHttpClient;
import calebzhou.rdi.core.server.utils.ThreadPool;
import com.mojang.brigadier.builder.LiteralArgumentBuilder;
import net.minecraft.commands.CommandSourceStack;
import net.minecraft.server.level.ServerPlayer;

/**
 * Created by calebzhou on 2022-09-29,19:59.
 */
public class InspectCommand extends RdiCommand {
	public InspectCommand() {
		super("inspect");
	}

	@Override
	public LiteralArgumentBuilder<CommandSourceStack> getExecution() {
		return baseArgBuilder.executes(context -> {
			ServerPlayer player = context.getSource().getPlayer();
			/*ThreadPool.newThread(()->{
				RdiHttpClient.
			});
*/

			return 1;
		});
	}
}

File: src/main/java/calebzhou/rdi/core/server/command/impl/InspectCommand.kt
package calebzhou.rdi.core.server.command.impl

import calebzhou.rdi.core.server.command.RdiCommand
import com.mojang.brigadier.builder.LiteralArgumentBuilder
import com.mojang.brigadier.context.CommandContext
import net.minecraft.commands.CommandSourceStack

/**
 * Created by calebzhou on 2022-09-29,19:59.
 */
class InspectCommand : RdiCommand("inspect") {
    override fun getExecution(): LiteralArgumentBuilder<CommandSourceStack> {
        return baseArgBuilder.executes { context: CommandContext<CommandSourceStack> ->
            val player = context.source.player
            1
        }
    }
}
